compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/transformers/FirCallCompletionResultsWriterTransformer.kt

```diff
@@ -189,7 +192,40 @@ class FirCallCompletionResultsWriterTransformer(
             }
             else -> {
                 resultType = typeRef.substituteTypeRef(subCandidate)
-                result.transformArguments(this, subCandidate.createArgumentsMapping()).transformExplicitReceiver(integerApproximator, null)
+                val vararg = subCandidate.argumentMapping?.values?.firstOrNull { it.isVararg }
+                result.transformArguments(this, subCandidate.createArgumentsMapping()).apply {
+                    if (vararg != null && this is FirFunctionCallImpl) {
+                        // Create a FirVarargArgumentExpression for the vararg arguments
+                        val resolvedArrayType = vararg.returnTypeRef.substitute(subCandidate)
+                        if (resolvedArrayType.typeArguments.size == 1) {
```

Comment:
This array might be primitive, so it will not have a generic parameter
`IntArray` for example, use `org.jetbrains.kotlin.fir.resolve.calls.ArgumentsKt#varargElementType`

Reply:
done